src/SFA.DAS.EmployerAccounts/Queries/GetAccountEmployerAgreementsRemove/GetAccountEmployerAgreementsRemoveQueryHandler.cs

```diff
@@ -57,45 +60,52 @@ public async Task<GetAccountEmployerAgreementsRemoveResponse> Handle(GetAccountE
+                removeEmployerAgreementView.HashedAgreementId =
+                    _hashingService.HashValue(removeEmployerAgreementView.Id);
+                removeEmployerAgreementView.HashedAccountId = message.HashedAccountId;
 
-                    if (result.Count == 1) continue;
-
-                    switch (removeEmployerAgreementView.Status)
-                    {
-                        case EmployerAgreementStatus.Pending:
-                            removeEmployerAgreementView.CanBeRemoved = true;
-                            break;
-                        case EmployerAgreementStatus.Signed:
+                var legalAgreementId = _hashingService.DecodeValue(removeEmployerAgreementView.HashedAgreementId);
 
-                            var commitmentConnectedToEntity = commitments.FirstOrDefault(c =>
-                                !string.IsNullOrEmpty(c.LegalEntityIdentifier)
-                                && c.LegalEntityIdentifier.Equals(removeEmployerAgreementView.LegalEntityCode)
-                                && c.LegalEntityOrganisationType == removeEmployerAgreementView.LegalEntitySource);
+                var agreement = await _employerAgreementRepository.GetEmployerAgreement(legalAgreementId);
 
-                            if (commitmentConnectedToEntity != null &&
-                                (commitmentConnectedToEntity.ActiveCount +
-                                 commitmentConnectedToEntity.PendingApprovalCount +
-                                 commitmentConnectedToEntity.PausedCount) != 0)
-                            {
-                                removeEmployerAgreementView.CanBeRemoved = false;
-                            }
-                            else
-                            {
-                                removeEmployerAgreementView.CanBeRemoved = true;
-                            }
+                var vacanciesSummary = await _dasRecruitService.GetVacanciesByLegalEntity(message.HashedAccountId, agreement.LegalEntityId);
+                    
+                if (result.Count == 1) continue;
 
-                            break;
-                        default:
+                switch (removeEmployerAgreementView.Status)
+                {
+                    case EmployerAgreementStatus.Pending:
+                        removeEmployerAgreementView.CanBeRemoved = true;
+                        break;
+                    case EmployerAgreementStatus.Signed:
+                        var commitmentConnectedToEntity = commitments.FirstOrDefault(c =>
+                            !string.IsNullOrEmpty(c.LegalEntityIdentifier)
+                            && c.LegalEntityIdentifier.Equals(removeEmployerAgreementView.LegalEntityCode)
+                            && c.LegalEntityOrganisationType == removeEmployerAgreementView.LegalEntitySource);
+
+                        if (vacanciesSummary.Vacancies.Any())
```

Comment:
The UI has a condition for having both vacancies and commitments but, unless I'm misreading, that's not possible given this code?